### ISSUE TITLE
non-coding-standard-compliant patterns in twisted.python.logger

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7712]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)| @wsanchez reported|
|-|-|
|Trac ID|trac#7712|
|Type|defect|
|Created|2014-11-11 01:11:48Z|
|Branch|https://github.com/twisted/twisted/tree/log-cleanup-7712|

As noted by exarkun in [#7596](https://github.com/twisted/twisted/issues/7596) (comment 12), there are some non-coding-standard-compliant patterns in `twisted.python.logger`:

1. assertIs (stdlib unittest spelling) is preferred over assertIdentical.
1. assertEqual is preferred over assertEquals (the latter is deprecated in the stdlib unittest module).
1. test_fooBar is preferred over test_foo_bar (_ means "dynamic dispatch here" in the Twisted coding standard).
1. Many test method docstrings say "should". The preferred style is "X does y" rather than "X should do y".

<details><summary>Searchable metadata</summary>

```
trac-id__7712 7712
type__defect defect
reporter__wsanchez wsanchez
priority__normal normal
milestone__New_Logging_System New Logging System
branch__branches_log_cleanup_7712 branches/log-cleanup-7712
branch_author__ 
status__closed closed
resolution__fixed fixed
component__logger logger
keywords__ 
time__1415668308686153 1415668308686153
changetime__1481165838395345 1481165838395345
version__None None
owner__wsanchez wsanchez
cc__glyph cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_1"></a>|@wsanchez commented|
|-|-|

Fixes on branch for [#7596](https://github.com/twisted/twisted/issues/7596):

1. r43085
1. r43086
1. r43087
1. r43093

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_2"></a>|@wsanchez commented|
|-|-|

Waiting on [#7710](https://github.com/twisted/twisted/issues/7710) and [#7711](https://github.com/twisted/twisted/issues/7711) before branching for this to avoid conflicts.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set owner to @wsanchez<br>@wsanchez set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_4"></a>|@wsanchez commented|
|-|-|

adiroiban mentions in [#7697](https://github.com/twisted/twisted/issues/7697) that try/finally around file handles in tests is noisy.  I thought that was listed here, but it's not, so noting it.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_6"></a>|@wsanchez commented|
|-|-|

Branching from branches/log-booyah-6750-5

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_7"></a>|@wsanchez commented|
|-|-|

r44389 changes `assertIdentical` → `assertIs`

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_8"></a>|@wsanchez commented|
|-|-|

r44390 changes `assertEquals` → `assertEqual`

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_9"></a>|@wsanchez commented|
|-|-|

r44391 fixes some conventional conventions about should and underbars.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_10"></a>|@wsanchez commented|
|-|-|

r44392 replaces `try`/`finally` around file use with `with`.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_11"></a>|@wsanchez commented|
|-|-|

to review

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez removed owner<br>@wsanchez set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_13"></a>|@ralphm set owner to @wsanchez|
|-|-|


 * In `test_io.LoggingFileTests`, there is a `try`/`finally` that could also be written with `self.patch` or `self.addCleanup`.
 * In `test_json` there are some `finally`s.
 * In `test_logger.TestLogger` you can also use `self.addCleanup` to do the observer removal.
 * In `test_filter` there are a bunch of methods with the prefix `test_should`. Why is there `should` in their name?

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_14"></a>|@wsanchez set owner to @ralphm|
|-|-|

Thanks.

* r44394 uses `self.patch()` and `self.addCleanup()`.
* r44395 fixes the missed `finally`→`with` change.
* `test_logger.TestLogger` isn't a test class, so doesn't have `self.addCleanup()`.
* The `test_shouldLogEvent*` methods are testing the logic re: whether an event should be logged by a filter.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_15"></a>|@ralphm set owner to @wsanchez|
|-|-|

Looks good to me. Please merge into the logger branch.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_16"></a>|@wsanchez set status to `closed`|
|-|-|

(In [44418]) Merge branches/log-cleanup-7712: coding-standard cleanup

Author: wsanchez
Reviewer: ralphm
Fixes: [#7712](https://github.com/twisted/twisted/issues/7712)
